docs/illink-tasks.md

```diff
@@ -0,0 +1,125 @@
+# ILLink.Tasks
+
+## Usage
+
+To enable ILLinker set `PublishTrimmed` property to `true` in your project and publish your app as a self-contained.
+
+```
+dotnet publish -r <rid> -c Release -p:PublishTrimmed=true
+```
+
+alternatively you can edit your .csproj file to include 
+
+```xml
+  <PropertyGroup>
+    <PublishTrimmed>true</PublishTrimmed>
+  </PropertyGroup>
+```
+
+The output will include only necessary code to run your application. The framework libraries size will be reduced noticeably.
+
+## ILLink Task Properties
+
+### ExtraArgs
+
+Additional [options](illink-options.md) passed to ILLinker.
+
+### OutputDirectory
+
+The directory in which to place processed assemblies.
+
+### ReferenceAssemblyPaths
+
+Assembly files with paths to assemblies needed as references.
+
+### RootAssemblyNames
+
+The names of the assemblies to root. This should contain assembly names without an extension, not file names or
+paths.
+
+### RootDescriptorFiles
+
+A list of XML [descriptors](data-formats.md#descriptor-format) files specifying linker roots at a granular level.
+
+## ILLink Task Customization
+
+The linker can be invoked as an MSBuild task, `ILLink`. We recommend not using the task directly, because the SDK has built-in logic that handles computing the right set of reference assemblies as inputs, incremental linking, and similar logic. If you would like to use the [advanced options](illink-options.md), you can invoke the msbuild task directly and pass any extra arguments like this:
+
+```xml
+<ILLink AssemblyPaths="@(AssemblyFilesToLink)"
+        RootAssemblyNames="@(LinkerRootAssemblies)"
+        RootDescriptorFiles="@(LinkerRootDescriptors)"
+        OutputDirectory="output"
+        ExtraArgs="-t -c link" />
+```
+
+## Default Linking Behaviour
+
+### .NET 5.0
+
+By default, the linker will operate in a full linking mode for all framework or 
+core managed assemblies. The 3rd party libraries or final app will be analyzed but not linked.
+This setting can be altered by setting  
```

Comment:
Unfinished sentence. I think the relevant option is `_TrimmerDefaultAction`. Setting it to `copyused` would switch to the 3.x behavior. It's also possible tell it to trim 3rd-party libraries, but it's not that convenient so we probably shouldn't document it as-is.

I think we should move most of the stuff under .NET 3.x to a common heading, and describe only the differences under net5.0/net3.x. The difference as far as I understand is that net5.0 will "link" fx assemblies, whereas net3.x does "copyused". The behavior for non-fx assemblies is the same.

Reply:
Finished the sentence. We should do more updates on this once the properties are "public"